src/PermissionsHelper.php

```diff
@@ -87,7 +85,18 @@ protected function userHas( $attributeOfUser, $attributeNameInDomElement ) {
 		}
 
 		$expectedValues = $this->getValueListFromAttribute( $attributeNameInDomElement );
-		$observedValues = call_user_func( $attributeAccessors[ $attributeOfUser ] );
+		switch ( $attributeOfUser ) {
+			case 'group':
+				$observedValues = MediaWikiServices::getInstance()
+					->getUserGroupManager()
+					->getUserEffectiveGroups( $user );
```

Comment:
Still checking if `getUserEffectiveGroups`'s implicit value for the second parameter (`$queryFlags = self::READ_NORMAL`) is fine. In the deprecated `getEffectiveGroups` method it is passing in `User->queryFlagsUsed` where the default is the same, but it is possible for that property's value to change (`User->load*`). However, I'm not 100% sure if the cases where it is changeable are relevant to Chameleon. It seems like when `User` is retrieved using `Skin->getUser()`) it doesn't call those load methods.

Reply:
I suspect this is fine